lib/Parse/ParseDecl.cpp

```diff
@@ -5950,6 +5950,9 @@ ParserStatus Parser::parseLineDirective(bool isLine) {
   // line.
   while (*LastTokTextEnd == ' ' || *LastTokTextEnd == '\t')
     ++LastTokTextEnd;
+  if (*LastTokTextEnd == '\r') {
```

Comment:
IMO this code should read
```
  while (*LastTokTextEnd == ' ' || *LastTokTextEnd == '\t')
    ++LastTokTextEnd;

  SourceLoc nextLineStartLoc;
  switch (*LastTokTextEnd++) {
  case '\r':
    if (*LastTokTextEnd == '\n')
      ++LastTokTextEnd;
    LLVM_FALLTHROUGH;
  case '\n':
    nextLineStartLoc = Lexer::getSourceLoc(LastTokTextEnd);
    break;
  default:
    diagnose(Tok.getLoc(), diag::extra_tokens_line_directive);
    return makeParserError();
  }
```

That better matches what we do in the lexer (namely, we generally regard `\r`, `\r\n` and `\n` as line endings; there are probably some omissions, but those are bugs IMO).